flink/v1.18/flink/src/test/java/org/apache/iceberg/flink/source/TestMetadataTableReadableMetrics.java

```diff
@@ -217,27 +217,27 @@ public void testPrimitiveColumns() throws Exception {
 
     Row binaryCol =
         Row.of(
-            52L,
+            55L,
             4L,
             2L,
             null,
             Base64.getDecoder().decode("1111"),
             Base64.getDecoder().decode("2222"));
-    Row booleanCol = Row.of(32L, 4L, 0L, null, false, true);
-    Row decimalCol = Row.of(85L, 4L, 1L, null, new BigDecimal("1.00"), new BigDecimal("2.00"));
-    Row doubleCol = Row.of(85L, 4L, 0L, 1L, 1.0D, 2.0D);
+    Row booleanCol = Row.of(36L, 4L, 0L, null, false, true);
+    Row decimalCol = Row.of(91L, 4L, 1L, null, new BigDecimal("1.00"), new BigDecimal("2.00"));
+    Row doubleCol = Row.of(91L, 4L, 0L, 1L, 1.0D, 2.0D);
```

Comment:
[optional] should we refactor this to pick file_size from the Datafiles themselve like we did we did in JDK 17 upgrade PR https://github.com/apache/iceberg/pull/7391#discussion_r1173288917

Never the less looks like size in bytes is increasing in this version is it because they are more accurate now ?

Reply:
Hey @singhpk234, that's an excellent suggestion. I've copied your approach here as well. Parquet now also tracks how large the data is in memory after compression (this is handy for strings where you don't know that upfront) so you can allocate buffers directly to the right size.

Reply:
> how large the data is in memory after compression (this is handy for strings where you don't know that upfront) so you can allocate buffers directly to the right size.

This is precisely what we needed in Redshift as well, our CBO was falling behind with variable length data types, will give them HeadsUp ! Thankyou @Fokko